enterprise/backend/src/metabase_enterprise/sandbox/query_processor/middleware/row_level_restrictions.clj

```diff
@@ -259,9 +260,24 @@
                      (query-perms/required-perms-for-query (:dataset-query (lib.metadata.protocols/card (qp.store/metadata-provider) card-id))
                                                  :throw-exceptions? true))
 
-    (let [table-ids (sandbox->table-ids sandbox)]
+    (let [table-ids (sandbox->table-ids sandbox)
+          table-id->db-id (if (seq table-ids)
+                            (m/index-by :id (t2/select :model/Table :id [:in table-ids]))
+                            {})
+          unblocked-table-ids (filter (fn [table-id] (data-perms/user-has-permission-for-table?
+                                                     api/*current-user-id*
+                                                     :perms/view-data
+                                                     :unrestricted
+                                                     (get table-id->db-id table-id)
+                                                     table-id))
+                                      table-ids)
+          create-queries-perms (zipmap unblocked-table-ids (repeat :query-builder))]
```

Comment:
Is there a reason to grant `:query-builder` to only the unblocked tables, but `:unrestricted` to all of them? I think it might be better to grant `:perms/view-data :unrestricted` to only the unblocked tables, and `:query-builder` to all of them.

Say you already have `:query-builder` access to all tables but are `:blocked` on all but one, where you have sandbox access. We don't want to override that block, right?

Reply:
I think you're right. I will switch that around and include it in the upcoming tests